corehq/apps/user_importer/importer.py

```diff
@@ -589,8 +589,8 @@ def create_or_update_commcare_users_and_groups(upload_domain, user_specs, upload
                     domain_info.group_memoizer.by_name(group_name).add_user(user, save=False)
 
             except ValidationError as e:
-                ret['errors'].append(e.message)
-            except (UserUploadError, CouchUser.Inconsistent) as e:
+                status_row['flag'] = e.message
+            except (UserUploadError, CouchUser.Inconsistent, ValidationError) as e:
```

Comment:
why catch this twice? I guess `str(e)` does get the message on the exception itself, so you should just need to add ValidationError to the else block present.
I guess that is what you intended but forgot to remove the block you added in the last commit

Reply:
Oops...this is what happens when you don't press CTRL+Z enough.

Reply:
@mkangia The reason for the separate catch is because `str(e)` results in `"['Invalid phone number detected']"` as opposed to just `'Invalid phone number detected'`. To get the latter result, `e.message` seems to be the way to go.

Reply:
How does this work for other exceptions? If flag is being set with a list of error messages for other exceptions as well, would be good to stick with that for consistency. I don't think these messages are shown to the user, are they?

Reply:
@mkangia Are you asking whether `status_row['flag']` is set with a list of errors? If so, no. `status_row['flag']` is set with one error, and `status_row` is then appended to `row`. (The errors are shown to the users).

Reply:
Offlining.